compiler/vm/vm.nim

```diff
@@ -965,47 +1214,82 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
       else:
         stackTrace(c, tos, pc, reportSem(rsemVmOverOrUnderflow))
     of opcLenSeq:
-      decodeBImm(rkInt)
+      decodeBImm(akInt)
       #assert regs[rb].kind == nkBracket
       let high = (imm and 1) # discard flags
       if (imm and nimNodeFlag) != 0:
+        # TODO: there should be a standalone opcode for NimNode.len
         # used by mNLen (NimNode.len)
-        regs[ra].intVal = regs[rb].node.safeLen - high
+        #regs[ra].intVal = regs[rb].node.safeLen - high
+        doAssert false
       else:
-        # safeArrLen also return string node len
-        # used when string is passed as openArray in VM
-        regs[ra].intVal = regs[rb].node.safeArrLen - high
+        {.warning: "What about openArray?".}
+        regs[ra].intVal = (regs[rb].atomVal.rawArrayVal.len /% regs[rb].atomVal.elemSize) - high
     of opcLenStr:
-      decodeBImm(rkInt)
-      assert regs[rb].kind == rkNode
-      regs[ra].intVal = regs[rb].node.strVal.len - imm
+      # a = b.len - imm
+      decodeBImm(akInt)
+      #[ TODO: as far as I can tell the, the `- imm` is only used for encoding magic mHigh (`high(string)`) in one opcode instead of two.
```

Comment:
How often does this occur in workloads the VM encounters is the question.

CISC vs RISC

Reply:
Yep, I really need some real world workload (the nim compiler iteself ;) ) to evaluate these proposals